### PR TITLE
ci(gha): run zombienet test only before runtime upgrade

### PR DESCRIPTION
**Pull request by 3h4x**
# Summary of changes

Rationale: zombienet is always run before runtime upgrade so any additional run on development is superfluous.
This will lower the pressure on our github runners

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

Release Notes:

- CI pipeline logic modified to run smoke test job only if changes are made to the source code and branch is not development.
- Superfluous test step removed from CI pipeline.

🎉 "Changes made with care, 
CI pipeline now aware. 
Smoke tests run with precision, 
Code quality on a mission." 🎉

### CODE REVIEW

**Comment by AwesomeIbex** [comment]
> # Summary of changes
> Rationale: zombienet is always run before runtime upgrade so any additional run on development is superfluous. This will lower the pressure on our github runners
> 
> # Acceptance Checklist
> What types of changes does your code introduce? _Put an `x` in the boxes that apply_
> 
> ## Dependencies
> * [ ]  Any dependent changes have been merged and published in downstream modules
> * [ ]  Any teams which manage the dependencies have been notified of the breaking changes
> 
> ## Bug (non-breaking change which fixes an issue):
> * [ ]  Have you **_written tests_** to protect against future occurrences of the bug?
> 
> ## Feature (non-breaking change which adds functionality)
> * [ ]  Have you **_seen it working_** in a development environment?
> * [ ]  Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
> * [ ]  Have you implemented this feature as per the **_issue acceptance criteria_**?
> 
> ### Substrate
> If this change involves substrate, have you remembered:
> 
> * [ ]  Storage Migration?
> * [ ]  RPC methods?
> * [ ]  Chainspec, both JSON specs & the module?
> * [ ]  Runtime versioning?
> * [ ]  Benchmarks?
> 
> ## Breaking change (a feature that would cause existing functionality not to work as expected)
> * [ ]  Is the breaking change **_documented_**?
> * [ ]  Are your commits fully representative of the change?
> * [ ]  Has any previous functionality been **_deprecated_** and versioned?
> 
> ## CI/CD
> * [ ]  If introducing new actions, have you **_looked at the action code_** for anything spurious?
> * [ ]  Have you written **_custom scripts_** for things that could be actions already on the marketplace?
> * [ ]  If introducing new actions, have you **_pegged a static version_**?
> 
> ## Documentation Update
> * [ ]  Have you checked other documentation, such as the docs repository?
> * [ ]  If updating script documentation, have you **_tested it on both environments_**?
> 
> ## Further comments
> Feel free to explain in further detail your choices if this is a significant change.
> 
> ## Screenshots (Please demonstrate this working in PolkadotJS)
> ### Summary by OpenAI
> Release Notes:
> 
> * CI pipeline logic modified to run smoke test job only if changes are made to the source code and branch is not development.
> * Superfluous test step removed from CI pipeline.
> 
> tada "Changes made with care, CI pipeline now aware. Smoke tests run with precision, Code quality on a mission." tada

We need smoke tests on every PR as nobody verifies if their pallet changes work or not.

**Review by AwesomeIbex — CHANGES_REQUESTED** [review]
See above

**Comment by 3h4x** [comment]
> We need smoke tests on every PR as nobody verifies if their pallet changes work or not.

This PR doesnt change that.

Please read my PR description explaining what this PR does:
> Rationale: zombienet is always run before runtime upgrade so any additional run on **development** is superfluous.

**Comment by AwesomeIbex** [comment]
> > We need smoke tests on every PR as nobody verifies if their pallet changes work or not.
> 
> This PR doesnt change that.
> 
> Please read my PR description explaining what this PR does:
> 
> > Rationale: zombienet is always run before runtime upgrade so any additional run on **development** is superfluous.

I read it, and I stand by my point.

If a PR does [skip-ci] we have no guarantees it works in a built blob until we try to runtime upgrade, passing the ownership of a PR passing this onto someone else.

**Comment by 3h4x** [comment]
> > > We need smoke tests on every PR as nobody verifies if their pallet changes work or not.
> > 
> > 
> > This PR doesnt change that.
> > Please read my PR description explaining what this PR does:
> > > Rationale: zombienet is always run before runtime upgrade so any additional run on **development** is superfluous.
> 
> I read it, and I stand by my point.
> 
> If a PR does [skip-ci] we have no guarantees it works in a built blob until we try to runtime upgrade, passing the ownership of a PR passing this onto someone else.

Let me throw more context in.

This PR has nothing to do with Merge Queue nor PR checks. It's literally single line of code adding condition to not run zombienet on development branch.
After merge we have squash merge commit, if there are changes defined in gha trigger
```
on:
  push:
    branches:
      - development
    paths:
      - "**/*.rs"
      - "**/Cargo.toml"
      - "**/Cargo.lock"
```
test-circuit workflow will run. This workflow runs tests and zombienet in two distinct jobs.

Additionally each merge that has such changes run `cargo release` which will create new commit that updates Cargo.toml with new versions.

After `cargo release`, new release is created and runtime upgrade initiated which first step is zombienet test.

Summing up zombienet will run on squash commit, cargo release commit and then before runtime upgrade. This is intended behaviour in your opinion?

**Review by AwesomeIbex — APPROVED** [review]
